docs/python-integration.md

```diff
@@ -0,0 +1,302 @@
+# SQLAlchemy Integration
+
+pgai provides SQLAlchemy integration for managing and querying vector embeddings in your database through a simple declarative interface.
+
+## VectorizerField
+
+### Basic Setup
+
+```python
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.orm import DeclarativeBase
+from pgai.extensions.sqlalchemy import VectorizerField
+from pgai.extensions import EmbeddingConfig
+
+class BlogPost(DeclarativeBase):
+    __tablename__ = "blog_posts"
+    
+    id = Column(Integer, primary_key=True)
+    title = Column(Text, nullable=False)
+    content = Column(Text, nullable=False)
+    
+    content_embeddings = VectorizerField(
+        source_column="content",
+        embedding=EmbeddingConfig(
+            model="text-embedding-3-small",
+            dimensions=768
+        )
+    )
+```
+
+### Querying with VectorizerField
+
+Once your model is set up, you can query the embeddings in several ways:
+
+#### Basic ORM Queries
+
+```python
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+# Get all embeddings
+with Session(engine) as session:
+    # Get embedding entries
+    embeddings = session.query(BlogPost.content_embeddings).all()
+    
+    # Access embedding vectors and chunks
+    for embedding in embeddings:
+        vector = embedding.embedding  # numpy array of embeddings
+        chunk = embedding.chunk      # text chunk that was embedded
+        chunk_seq = embedding.chunk_seq  # sequence number of chunk
+```
+
+#### Semantic Search
+
+Using pgvector's distance operators in queries:
+
+```python
+from sqlalchemy import func
+
+def search_similar_content(session: Session, query_text: str, limit: int = 5):
+    # Generate embedding for search query
+    return (
+        session.query(
+            BlogPost.content_embeddings,
+            BlogPost.title,
+            # Calculate and include distance in results
+            BlogPost.content_embeddings.embedding.cosine_distance(
+                func.ai.openai_embed(
+                    'text-embedding-3-small',
+                    query_text,
+                    text('dimensions => 768')
+                )
```

Comment:
Can we reference this from the Vectorizer config? That way we can remove the cognitive load on the user. Maybe that's what @cevian was referring when talking about improving the dev experience.

Reply:
This function is just plain sqlalchemy core right now, this works even without my extension code. We can add custom sql alchemy functions. e.g. so you don't have to wrap the dimension paramter in a `text`, I guess I could also run a subquery maybe in such a function. We can definitely provide a python helper that references the underlying vectorizer or loads it from the python config.

Mats idea was to do this on sql level though and then it's automatically also available in python so we safe some effort there.

Reply:
I agree we can improve this UX. But I also think we can do that in a separate PR